sklearn/metrics/tests/test_pairwise.py

```diff
@@ -893,3 +896,33 @@ def test_check_preserve_type():
                                                    XB.astype(np.float))
     assert_equal(XA_checked.dtype, np.float)
     assert_equal(XB_checked.dtype, np.float)
+
+
+@pytest.mark.parametrize("n_jobs", [1, 2, -1])
+@pytest.mark.parametrize("metric", ["seuclidean", "mahalanobis"])
+@pytest.mark.parametrize("dist_function",
+                         [pairwise_distances, pairwise_distances_chunked])
+@pytest.mark.parametrize("y_is_x", [True, False], ids=["Y==X", "Y!=X"])
+def test_pairwise_distances_data_derived_params(n_jobs, metric, dist_function,
+                                                y_is_x):
+    # check that pairwise_distances give the same result in sequential and
+    # parallel, when metric has data-derived parameters.
+    set_config(working_memory=0.1)  # to have more than 1 chunk
+
+    rng = np.random.RandomState(0)
+
+    X = rng.random_sample((1000, 10))
+
+    if y_is_x:
+        Y = X
+        if n_jobs == 1:
+            expected_dist = squareform(pdist(X, metric=metric))
+        else:
+            expected_dist = cdist(X, Y, metric=metric)
```

Comment:
Maybe use cdist in either case? We don't care much about performance in this test. And it would alow to simplify this test significantly.

Reply:
Well actually cdist and pdist disagree for mahalanobis distance on scipy [(comment)](https://github.com/scikit-learn/scikit-learn/issues/12672#issuecomment-443217075) and we use pdist when `n_jobs == 1` and cdist otherwise in sklearn, so I'm forced to make the distinction.

It would be better to only use one of them in sklearn.